File: src/economy/gardenMarketActivity.ts
import { FOOD_SALE_GOLD_PER_UNIT } from '../generated/gameBalance.ts';

export type GardenMarketActivityDef = {
  goldPerPersonPerSec: number;
  foodPerPersonPerSec: number;
  foodSelfShare: number;
};

/** Taxable market activity from a garden over a time window (seconds). */
export function gardenMarketActivity(
  def: GardenMarketActivityDef,
  population: number,
  seconds: number,
): number {
  const pop = Math.max(0, population);
  let activity = def.goldPerPersonPerSec * pop * seconds;

  if (def.foodPerPersonPerSec > 0) {
    const totalFood = def.foodPerPersonPerSec * pop * seconds;
    const soldFood = totalFood * (1 - def.foodSelfShare);
    activity += soldFood * FOOD_SALE_GOLD_PER_UNIT;
  }

  return activity;
}

export { SECONDS_PER_DAY } from '../world/gameCalendar.ts';
export { GAME_DAY_SECONDS } from '../world/gameCalendar.ts';

File: server/src/economy/garden_market_activity.rs
use crate::balance_generated::{BackyardGardenDef, FOOD_SALE_GOLD_PER_UNIT};

/// Taxable market activity from a garden over a time window (seconds).
pub fn garden_market_activity(def: &BackyardGardenDef, population: f64, seconds: f64) -> f64 {
    let population = population.max(0.0);
    let mut activity = def.gold_per_person_per_sec * population * seconds;

    if def.food_per_person_per_sec > 1e-9 {
        let total_food = def.food_per_person_per_sec * population * seconds;
        let sold_food = total_food * (1.0 - def.food_self_share.clamp(0.0, 1.0));
        activity += sold_food * FOOD_SALE_GOLD_PER_UNIT;
    }

    activity
}
